compiler/sigmatch.nim

```diff
@@ -633,20 +633,22 @@ proc procTypeRel(c: var TCandidate, f, a: PType): TTypeRelation =
   else: discard
 
 proc typeRangeRel(f, a: PType): TTypeRelation {.noinline.} =
-  let
-    a0 = firstOrd(a)
-    a1 = lastOrd(a)
-    f0 = firstOrd(f)
-    f1 = lastOrd(f)
-  if a0 == f0 and a1 == f1:
-    result = isEqual
-  elif a0 >= f0 and a1 <= f1:
-    result = isConvertible
-  elif a0 <= f1 and f0 <= a1:
-    # X..Y and C..D overlap iff (X <= D and C <= Y)
-    result = isConvertible
-  else:
-    result = isNone
+  template checkRange[T](afirst, alast, ffirst, flast: T): TTypeRelation = 
+    if afirst == ffirst and alast == flast:
+      isEqual
+    elif afirst >= ffirst and alast <= flast:
+      isConvertible
+    elif afirst <= flast and ffirst <= a_last:
+      # X..Y and C..D overlap iff (X <= D and C <= Y)
+      isConvertible
+    else:
+      isNone
+  
+  if f.isOrdinalType: 
```

Comment:
Move this condition out of ``typeRangeRel``. The callers should know which variant is used, there is no need to check for ``f.isOrdinalType`` at runtime.

Reply:
Not sure about this one.  Unless I got it completely wrong, ``typeRangeRel`` used only once and it is in ``typeRel``. It doesn't look like ``typeRel`` in tyRange branch knows the range type, so the same if statement will be required in ``typeRel``. Given that ``typeRel`` is huge, I would rather move if statement outside of ``typeRel``.

Let me know what you think.